### PR TITLE
Add motd docs

### PR DESCRIPTION
**Pull request by Shadowsniper784**
add docs for unconnected ping and pong as the motd

### CODE REVIEW

**Comment by theaddonn** [comment]
THAT WAS FAST!🔥

File: docs/MOTD.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+The MOTD follows a semicolon-separated (`;`) format comprising various fields, semi colons can be escaped with a backslash (`\`)
+```
+<Edition>;<MOTD>;<Protocol Version>;<Game Version>;<Player Count>;<Max Player Count>;<Server GUID>;<World Name>;<Game Mode>;<Nintendo Limited>;<IpV4Port>;<IpV6Port>;<Extra Data>;
+```
+
+#### Field Breakdown & Data Types
+
+- **Edition:** string  
+  Specifies the game edition (e.g., `MCPE` for Minecraft Pocket Edition), seems to be hardcoded to `MCPE`
```
Comment (theaddonn) [inline docs/MOTD.md:64]: I'm not actually sure about that one, it might be that Education Edition uses `MCEE` instead of `MCPE`

Comment (theaddonn) [reply docs/MOTD.md:64]: wiki.vg seems to say the same... 
image

File: docs/MOTD.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+- **MOTD:** string  
+  The actual message that is displayed, it can have Minecraft colors.  
+- **Protocol Version:** uint16  
+  Numeric identifier indicating the game protocol version (e.g., `748`).  
+- **Game Version:** string  
+  Denotes the server's game version (e.g., `1.20.40`, or on CubeCraft: `1`).
```
Comment (theaddonn) [inline docs/MOTD.md:70]: not directly an issue, but love how you mentioned it XD
I found the same thing a few months ago, it is quite hilarious... wonder if there is a good reason for it

File: docs/MOTD.md [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+- **port v4:** uint16  
+  The port of the server for IPv4, useful for servers that have both IPv4 and IPv6.  
+- **port v6:** uint16  
+  The port of the server for IPv6, useful for servers that have both IPv4 and IPv6.  
+- **Extra Data:**  
+  Ends with `0` to signify the conclusion of the MOTD string.  
+
+## Code Examples
+
+::: code-group
```
Comment (theaddonn) [inline docs/MOTD.md:92]: wonder if there is a way to make this collapsible, since not everyone might want to see it directly..

File: docs/MOTD.md [hunk 4]
```diff
@@ -0,0 +1,362 @@
+# MOTD Protocol Documentation
```
Comment (theaddonn) [inline docs/MOTD.md:1]: maybe add what motd directly stands for.. (Message of the day)

File: docs/MOTD.md [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+
+The server operates by listening on port `19132` for incoming RakNet unconnected pings. Upon receiving a valid ping, the server responds with an unconnected pong that contains the MOTD and other info. This interaction ensures that clients receive essential details about the server before establishing a connection.
+
+## Packet Structure
+
+### Unconnected Ping
+
+When a client wants to display the server it sends an unconnected ping
+
+- **Packet ID (1 byte):** `0x01`
```
Comment (theaddonn) [inline docs/MOTD.md:13]: I wonder in general if we should use a schema like rust uses (`u32`) or more over the older C style (`unsigned int`) or something in between (`uint32`). I personally prefer something like `u32`...

**Review by theaddonn — CHANGES_REQUESTED** [review]
Seems very good! But there are still a few things that might need an improvement..

oh, also move this into the [`docs/raknet/`](https://github.com/bedrock-crustaceans/protocol-wiki/tree/main/docs/raknet) directory, thanks :)

File: docs/MOTD.md [hunk 6]
```diff
@@ -0,0 +1,362 @@
+# MOTD Protocol Documentation
+
+## Overview
+
+The server operates by listening on port `19132` for incoming RakNet unconnected pings. Upon receiving a valid ping, the server responds with an unconnected pong that contains the MOTD and other info. This interaction ensures that clients receive essential details about the server before establishing a connection.
```
Comment (MisledWater79) [inline docs/MOTD.md:5]: Maybe rephrase to be:
```suggestion
A RakNet server, by default, listens on port `19132` for incoming [Unconnected Pings](#unconnected-ping). Once a valid ping is sent, the server sends back a [Unconnected Pong](#unconnected-pong). This contains the server's (MOTD)[#motd-format), along with other server info. All this ensures that a client receives the details of the server before a connection is established.
```

**Review by MisledWater79 — CHANGES_REQUESTED** [review]

**Comment by theaddonn** [comment]
How are the changes going?

**Comment by MisledWater79** [comment]
Actually scratch that, thats a LOT of code... maybe move that to a link to github or different page.

File: docs/raknet/MOTD.md [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+```
+:::
+## Sources
+- [Go Raknet](https://github.com/Sandertv/go-raknet/blob/master/internal/message/unconnected_ping.go#L8)<br>
+- [PHP MC Query](https://github.com/xPaw/PHP-Minecraft-Query/blob/master/src/MinecraftQuery.php#L212)<br>
+- [Raknet](https://github.com/facebookarchive/RakNet/blob/1a169895a900c9fc4841c556e16514182b75faf8/Source/RakPeer.cpp#L135)<br>
+- [Phantom](https://github.com/jhead/phantom/blob/44056d83afbfe60b253faff01308d171ac21e8d6/internal/proto/proto.go#L23)<br>
+- [Geyser](https://github.com/GeyserMC/Protocol/blob/46b4ad37b159b1fc59e45871f7572101f5ed43ab/bedrock-connection/src/main/java/org/cloudburstmc/protocol/bedrock/BedrockPong.java#L23)<br>
+- [Wiki.vg](https://wiki.vg/Raknet_Protocol#Unconnected_Ping)<br>
+- Wireshark
```
Comment (MisledWater79) [inline docs/raknet/MOTD.md:364]: Use `#### Resources` instead, along with that set the links to be this format: `[[wiki.vg/Minecraft Wiki](https://minecraft.wiki/w/Minecraft_Wiki:Projects/wiki.vg_merge/Raknet_Protocol)]: Some description here of what was used` (I think the wiki.vg link is the only one that needs updated since it was moved to the actual wiki)

File: docs/raknet/MOTD.md [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+  Returns the status of Nintendo's limitation to join (0 or 1).  
+- **port v4:** uint16  
+  The port of the server for IPv4, useful for servers that have both IPv4 and IPv6.  
+- **port v6:** uint16  
+  The port of the server for IPv6, useful for servers that have both IPv4 and IPv6.  
+- **Extra Data:**  
+  Ends with `0` to signify the conclusion of the MOTD string.  
+
+## Code Examples
+::: details Code Examples
```
Comment (MisledWater79) [inline docs/raknet/MOTD.md:91]: Remove the dropdown as ill be implementing it directly to the code groups.

File: docs/raknet/MOTD.md [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+1. **Packet Size:**  
+   The total size must be exactly 33 bytes.
+2. **Packet ID:**  
+   Must strictly be `0x01`.
+3. **Timestamp:**  
+   It is just a timestamp of uptime, but we can check it is not infront of unix timestamp
+4. **Magic Offline Message ID:**  
+   Must match the predefined 16-byte sequence.
+5. **GUID:**  
+   It doesn't really need to be checked, but it is there
```
Comment (MisledWater79) [inline docs/raknet/MOTD.md:37]: I don't see a point to this area, personally I think it's not really useful. let me know if I'm wrong though.

File: docs/raknet/MOTD.md
Comment (MisledWater79) [inline docs/raknet/MOTD.md:1]: The explanaition of the packets shouldn't be in this file, as this is labeled `Message of the Day (MOTD) Protocol Documentation` so it should be more just on the motd and not explaining the entirety of where it comes from.

**Review by MisledWater79 — CHANGES_REQUESTED** [review]

File: docs/raknet/MOTD.md [hunk 10]
```diff
@@ -0,0 +1,364 @@
...
+- **port v6:** uint16  
+  The port of the server for IPv6, useful for servers that have both IPv4 and IPv6.  
+- **Extra Data:**  
+  Ends with `0` to signify the conclusion of the MOTD string.  
+
+## Code Examples
+::: details Code Examples
+::: code-group
+
+```rust [src/main.rs]
```
Comment (MisledWater79) [inline docs/raknet/MOTD.md:94]: I'd say change the names to just be a Rust and Typescript instead of a file path.

File: docs/raknet/MOTD.md
Comment (Shadowsniper784) [reply docs/raknet/MOTD.md:1]: Should I split it into separate files like: unconnectedpong.md or something like that? Maybe in like a MOTD subgroup as they aren't used elsewhere

Comment (MisledWater79) [reply docs/raknet/MOTD.md:1]: I would say keep all motd stuff in this file and either move it to and make the unconnected pong file, or just remove it and someone else could make it and reference this file
